src/formatDate.js

```diff
@@ -50,7 +50,44 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const previousSeparator = fromFormat[3];
+  const previousDate = date.split(previousSeparator);
+  const currentSeparator = toFormat[3];
+  const currentDate = [];
+
+  let previousYearFormat = '';
+  let day = '';
+  let month = '';
+  let year = '';
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'DD') {
+      day = previousDate[i];
+    } else if (fromFormat[i] === 'MM') {
+      month = previousDate[i];
+    } else if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      previousYearFormat = fromFormat[i];
+      year = previousDate[i];
+    }
+  }
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    if (toFormat[i] === 'DD') {
+      currentDate.push(day);
+    } else if (toFormat[i] === 'MM') {
+      currentDate.push(month);
+    } else if (toFormat[i] === previousYearFormat) {
+      currentDate.push(year);
+    } else if (toFormat[i] === 'YY' && previousYearFormat === 'YYYY') {
+      currentDate.push(year.slice(-2));
+    } else {
+      const getFullYear = year < 30 ? '20' : '19';
+
+      currentDate.push(getFullYear + year);
```

Comment:
let's create a function for the correct year formatting

Reply:
done